s3fs/core.py

```diff
@@ -1814,14 +1814,49 @@ def __init__(
         self.append_block = False
 
         if "a" in mode and s3.exists(path):
-            loc = s3.info(path)["size"]
+            head = {
+                k: v
+                for k, v in self._call_s3(
```

Comment:
The previous version allowed the info to come from cache, but here we explicitly repeat. This might be OK, or maybe we should include the header keys that matter in the normal info, so we can reuse them.

Reply:
It's still something I can't find best way to reflect, on one hand it's nice to preserve it in cache, but heavy to get all keys

Since it's mainly used on append, I think it might be better to reflect it one time at initialization and let user cache it or not later